add some vargo options: --vargo-verbose, --vstd-no-verify, --color

tjhance opened the pull request:
- `--vargo-verbose` - outputs all the commands that vargo runs, should be useful for debugging
- `--vstd-no-verify` - skips verification of vstd
- forward `--color` to cargo (this is useful to me because i like to pipe the output into `head`)

There's one thing I couldn't get to work that I need help with. Passing `--no-verify` to rust_verify doesn't seem to work, I just get `error: Unrecognized option: 'no-verify'`. I think it's because of `--internal-build-vstd-driver`? But I'm not sure the best way to fix it.

utaal commented:
I changed this to forward `--color arg`, instead of `--color=arg`. `cargo` allows both; we could also allow both. Let me know if you have a preference, or if my commit doesn't look right.

utaal commented:
> There's one thing I couldn't get to work that I need help with. Passing --no-verify to rust_verify doesn't seem to work, I just get error: Unrecognized option: 'no-verify'. I think it's because of --internal-build-vstd-driver? But I'm not sure the best way to fix it.

I think https://github.com/verus-lang/verus/pull/513/commits/1fe6809cc561fb0a14b7939927059fbb2d412c75 should do the trick. It works for me. Please check that it matches what you intended.

utaal commented:
I think it'd be nice if `-v`, `-vv`, and `--verbose` would also implicitly enable `--vargo-verbose`.

utaal commented on source/vstd_build/src/main.rs:
```diff
@@ -77,10 +92,19 @@ fn main() {
-        .args(&child_args[..])
-        .spawn()
-        .expect("could not execute lifetime rustc process");
+    let mut child = std::process::Command::new(cmd);
+    child.args(&child_args[..]);
+
+    if verbose {
+        // This environment variable (set by cargo) is important
+        // Put it on the command explicitly so it shows up in the verbose output
+        if let Ok(path) = std::env::var("DYLD_FALLBACK_LIBRARY_PATH") {
+            child.env("DYLD_FALLBACK_LIBRARY_PATH", path);
+        }
+        log_command(&child);
+    }
```
(If that sounds good to you, I can take care of adding `vargo run`; I am already doing something similar for https://github.com/verus-lang/verus/pull/522)

tjhance replied:
I don't have a huge preference. But I did find it informative the way it is (part of the reason I added this in the first place was to understand exactly how the rust-verify binary gets called)

tjhance commented:
I added support for more verbose flags. I also just removed the PATH stuff since it sounds like you're making changes to the way all that works anyway.

tjhance commented:
@utaal are there are any more issues on this?

tjhance commented:
(also pushed to main)
